test/product-import.spec.coffee

```diff
@@ -510,6 +510,19 @@ describe 'ProductImport unit tests', ->
         done()
       .catch done
 
+    it 'should throw error in case of too big query', (done) ->
+      exptectedErrorMessage = 'product fetch query size: 10952 bytes, exceeded the supported size, please try with a smaller batch size.'
```

Comment:
Then your query is wrong.
A) Either I search all skus in only the master variant
B) you query for only the master sku in all variants

The URL length of B is almost constant, while A still grows with the number of variant.

B is implemented in the csv sync.